src/util/ser.rs

```diff
@@ -0,0 +1,373 @@
+use std::result::Result;
+use std::io::{self, Read, Write};
+use std::fmt;
+use std::collections::HashMap;
+use std::hash::Hash;
+use std::mem;
+
+use secp256k1::Secp256k1;
+use secp256k1::key::PublicKey;
+use bitcoin::blockdata::script::Script;
+use bitcoin::util::hash::Sha256dHash;
+use std::marker::Sized;
+
+pub const MAX_VEC_SIZE: usize = 32 * 1024 * 1024;
+
+pub enum Error {
+	CorruptedData,
+	BadPublicKey,
+	IoError(io::Error),
+	InvalidLength,
+	InvalidValue,
+}
+
+impl fmt::Debug for Error {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match *self {
+			Error::CorruptedData => f.write_str("corrupted data"),
+			Error::BadPublicKey => f.write_str("Bad public key"),
+			Error::IoError(ref e) => write!(f, "{}", e),
+			Error::InvalidLength => f.write_str("Invalid length"),
+			Error::InvalidValue => f.write_str("Invalid value"),
+		}
+  }
+}
+
+impl From<io::Error> for Error {
+  fn from(e: io::Error) -> Self {
+		Error::IoError(e)
+  }
+}
+
+pub struct Writer<W> { writer: W }
+pub struct Reader<R> { reader: R }
+
+pub trait Writeable<W: Write> {
```

Comment:
Probably bikeshedding, but why don't give it a name like `Encodable`/`Decodable` just like how `MsgEncodable`/`MsgDecodable` are named?